### PR TITLE
Fuzz Chunked and Struct arrays

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

**Comment by robert3005** [comment]
This uncovers some new and surprising bugs

**Comment by robert3005** [comment]
But after this we have coverage of all canonical arrays (apart from Null but that's a boring case)

**Comment by robert3005** [comment]
...aaaand extensions

**Review by a10y — APPROVED** [review]

**Comment by robert3005** [comment]
I will merge this and start tackling the fuzz failures. Don't want to grow this pr too big

File: vortex-array/src/array/varbinview/mod.rs [hunk 1]
```diff
@@ -223,7 +223,7 @@ impl VarBinViewArray {
         for b in iter {
             builder.append_value(b);
         }
-        let array = Array::from_arrow(&builder.finish(), true);
+        let array = Array::from_arrow(&builder.finish(), false);
```
Comment (robert3005) [inline vortex-array/src/array/varbinview/mod.rs:226]: there was one real bug that I found while making fuzzing changes

**Pull request merged**